scripts/bootloader/fix_size.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+
+
+import sys
+import argparse
+import struct
+import re
+from intelhex import IntelHex
+from collections import namedtuple
+
+# See fw_info.h
+Fwinfo = namedtuple('Fwinfo',
+                    'magic total_size size version address boot_address '
+                    'valid reserved ext_api_num ext_api_request_num')
+FW_INFO_FSTR = '<12sIIIIII16sII'
+FW_INFO_SZ = struct.calcsize(FW_INFO_FSTR)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("-m", "--magic-value", required=True,
+                        help="ASCII representation of magic value.")
+    parser.add_argument("-i", "--in", "-in", required=True, dest="infile",
+                        type=argparse.FileType('rb'),
+                        help="Input hex file to update 'size' fields in.")
+    parser.add_argument("-o", "--out", "-out", required=False, dest="outfile",
+                        type=argparse.FileType('wb'),
+                        help="Output hex file to write result to.")
+    parser.add_argument("--offset", required=True, type=lambda x: int(x, 0),
+                        help="Offset of fw_info inside image.")
+    parser.add_argument("--valid-val", required=True, type=lambda x: int(x, 0),
+                        help="Value of 'valid' field.")
+
+    return parser.parse_args()
+
+
+if __name__ == "__main__":
+    args = parse_args()
+    h = IntelHex(args.infile.name)
+    bs = h.tobinstr()
+    magic = b''.join([struct.pack("<I", int(m, 0))
+                      for m in args.magic_value.split(",")])
+    magic_addresses = [m.start()
```

Comment:
This actually messes things a bit, since I use the `addr` field to index into the `bs` binstr.

If the address in `magic_addresses` is absolute, then we need `bs` to start from 0, which does not make sense, since the hex file does  not start at address 0.

We can get around this by subtracting the `hex_file_base` when indexing into `bs` but IMO this is a inferior solution.

Reply:
Ah, right. I'd prefer if the names were changed a bit to not give the impression that they are addresses. Maybe use "index" or something?

Reply:
Changed to `magic_index_list` and `idx`